pytorch_lightning/loggers/base.py

```diff
@@ -340,6 +340,19 @@ def _add_prefix(self, metrics: Dict[str, float]):
 
         return metrics
 
+    def on_save_checkpoint(self, trainer, pl_module: LightningModule, checkpoint: Dict[str, Any]) -> dict:
```

Comment:
I propose to change this into

```
@property
def state_dict(self):
     pass
```
or similar.

This is not like a callback method, and I don't see a reason why the logger needs access to the trainer, pl_module or why it would modify the existing checkpoint.